astropy/coordinates/builtin_frames/galactocentric.py

```diff
@@ -88,76 +98,95 @@ class galactocentric_frame_defaults(ScienceState):
-        params['galcen_coord'] = ICRS(ra=266.4051*u.degree,
-                                      dec=-28.936175*u.degree)
-        references['galcen_coord'] = \
-            'http://adsabs.harvard.edu/abs/2004ApJ...616..872R'
-
-        # The roll angle is the same for both frames:
-        params['roll'] = 0 * u.deg
-
-        if arg == 'pre-v4.0':
-            params['galcen_distance'] = 8.3 * u.kpc
-            references['galcen_distance'] = \
-                'https://ui.adsabs.harvard.edu/#abs/2009ApJ...692.1075G'
-
-            params['galcen_v_sun'] = r.CartesianDifferential([11.1,
-                                                              220+12.24,
-                                                              7.25]*u.km/u.s)
-            references['galcen_v_sun'] = \
-                ['https://ui.adsabs.harvard.edu/#abs/2010MNRAS.403.1829S',
-                 'https://ui.adsabs.harvard.edu/#abs/2015ApJS..216...29B']
-
-            params['z_sun'] = 27.0 * u.pc
-            references['z_sun'] = \
-                'https://ui.adsabs.harvard.edu/#abs/2001ApJ...553..184C'
-
-        elif arg == 'v4.0':
-            params['galcen_distance'] = 8.122 * u.kpc
-            references['galcen_distance'] = \
-                'https://ui.adsabs.harvard.edu/abs/2018A%26A...615L..15G'
-
-            params['galcen_v_sun'] = r.CartesianDifferential([12.9,
-                                                              245.6,
-                                                              7.78]*u.km/u.s)
-            references['galcen_v_sun'] = \
-                ['https://ui.adsabs.harvard.edu/abs/2018RNAAS...2..210D',
-                 'https://ui.adsabs.harvard.edu/abs/2018A%26A...615L..15G',
-                 'https://ui.adsabs.harvard.edu/abs/2004ApJ...616..872R']
-
-            params['z_sun'] = 20.8 * u.pc
-            references['z_sun'] = \
-                'https://ui.adsabs.harvard.edu/abs/2019MNRAS.482.1417B'
-
+        if arg in cls._registry:
+            state = cls._registry[arg].copy()
+            params = dict(state.get("value"))  # ensure mutable
+            references = dict(state.get("references", dict()))
         else:
-            raise ValueError(f'Invalid string input to retrieve solar '
-                             f'parameters for Galactocentric frame: "{arg}"')
+            raise ValueError(
+                f"Invalid string input to retrieve solar "
+                f'parameters for Galactocentric frame: "{arg}"'
+            )
 
-        return params, references
+        return params, references, state
 
     @classmethod
     def validate(cls, value):
         if value is None:
             value = cls._latest_value
 
         if isinstance(value, str):
-            params, refs = cls.get_solar_params_from_string(value)
+            params, refs, state = cls.get_solar_params_from_string(value)
             cls._references = refs
+            cls._state = state
```

Comment:
I wonder if we shouldn't simply store the parameters as well.

Reply:
I think this should be in the other PR.